File: backend/app/api/v1/schemas/recommend.py
from pydantic import BaseModel
from typing import List, Optional


class RecommendRequest(BaseModel):
    genres: List[str]
    vibes: List[str]
    themes: List[str]
    pacePreference: Optional[str] = None
    lengthPreference: Optional[str] = None
    limit: int = 5


class BookResponse(BaseModel):
    id: str
    title: str
    author: str
    description: Optional[str]
    genres: List[str]
    pages: Optional[int]
    cover_url: Optional[str]
    score: float

File: frontend/src/api/recommend.ts
export type RecommendRequest = {
  user_id: string;
  genres: string[];
  vibes: string[];
  themes: string[];
  pacePreference?: string | null;
  lengthPreference?: string | null;
  limit?: number;
};

export type RecommendedBook = {
  id: string;
  title: string;
  author: string;
  description?: string;
  genres: string[];
  pages?: number;
  cover_url?: string;
  score: number;
  reasons: string[];
};

export async function fetchRecommendations(
  payload: RecommendRequest
): Promise<RecommendedBook[]> {
  const res = await fetch("http://127.0.0.1:8000/api/v1/recommend", {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify(payload),
  });

  if (!res.ok) {
    throw new Error("Failed to fetch recommendations");
  }

  return res.json();
}
